src/psyclone/psyGen.py

```diff
@@ -5967,10 +5968,27 @@ class CodeBlock(Node):
     :param statements: list of fparser2 AST nodes representing the Fortran \
                        code constituting the code block.
     :type statements: list of :py:class:`fparser.two.utils.Base`
+    :param structure: optional argument indicating whether this code \
+    block is a statement or an expression.
+    :type structure: :py:class:`psyclone.psyGen.CodeBlock.Structure` \
+    or `NoneType`
     :param parent: the parent node of this code block in the PSyIR.
     :type parent: :py:class:`psyclone.psyGen.Node`
+
     '''
-    def __init__(self, statements, parent=None):
+    class Structure(Enum):
+        '''
+        Enumeration that captures the structure of the code block which
+        may be required when processing.
+
+        '''
+        # The Code Block comprises one or more Fortran statements
+        # (which themselves may contain expressions).
+        STATEMENT = 1
+        # The Code Block comprises one or more Fortran expressions.
+        EXPRESSION = 2
+
+    def __init__(self, statements, structure=None, parent=None):
```

Comment:
The `statements` and `self._statements` should probably be renamed to something else (nodes?, fp2_nodes?), as they are not just statements. This applies to all the `CodeBlock` class, including docstrings.

Reply:
OK, I think I've done this. I also modified the names in the nodes_to_code_block function and noticed that the docstrings were not correct so fixed those too.